docs/terminology/rules-validation.md

```diff
@@ -0,0 +1,109 @@
+# Rules Validation
+
+**NEAT** supports the validation of the consistency of a `Rule` object as a whole. This includes
+checking the `Rule` object against previous iteration and another `Rule` object
+that is the basis for the current `Rule` object.
+
+There are three fields in the `Metadata` of a `Rule` object that tells **NEAT** how to validate the `Rule` object:
+
+* `dataModelType` - The type of data model the `Rule` object represents. This can be either `solution` or `enterprise`.
+* `schema` - The schema completeness of the `Rule` object. This can be either `complete`, `extension`, or `partial`.
+* `extension` - In the case of `schema=extension`, this field specifies how the `Rule` object extends the previous iteration.
+
+There can be up to three different `Rule` objects that are used for validation. In a spreadsheet, these have
+different prefixes:
+
+* **No prefix**: The main `Rules` object often referred to as the `user` `Rules` object.
+* **`Last`**: The previous iteration of the `user` `Rules` object.
+* **`Ref`**: (short for Reference) A `Rules` object that is referenced by the `user` `Rules` object.
+
+## <code>dataModelType</code>—Whether to Validate against Reference Object
+
+If the `dataModelType` is set to `enterprise`, then the `Rule` object is expected to be a fundamental data model,
+meaning that is not based on another data model. In this case, if there is a `Reference` it will be ignored.
+
+If the `dataModelType` is set to `solution`, then the `Rule` object is expected to be based on another data model.
+For example, the user rules might use classes, views, or containers from the reference rules. If the `Reference` rules
+is missing an error will be raised.
+
+## <code>schema</code>—Whether to Validate against Last Rules object
+
+If the `schema` is set to `complete`, then the `Rule` object is expected to be a complete data model. This means that
+the `Rule` object should contain all the classes, views, and containers that are needed for the data model.
```

Comment:
data model = Solution
schema = complete
=> user + reference === fully complete (in case when we have mapping to enterprise data model)